### PR TITLE
Contacts in extension foundation

### PR DESCRIPTION
**Pull request by zatteo**
- Contact cipher type and related classes
- Contact query
- Contact list
- Contact links (redirect to "Contacts" app on open contact, edit contact, ...)

image

### CODE REVIEW

File: libs/cozy/queries.ts [hunk 1]
```diff
@@ -78,3 +78,39 @@ export const fetchPaper = async (client: CozyClient, _id: string) => {
...
+        },
+        {
+          trashed: false,
+        },
+      ],
+    })
+    .indexFields(["_id"])
+    .limitBy(1000),
+  options: {
+    as: "contacts",
```
Comment (Crash--) [inline libs/cozy/queries.ts:106]: `as: io.cozy.contacts` https://github.com/cozy/cozy-guidelines/?tab=readme-ov-file#naming-of-queries

File: apps/browser/src/cozy/components/add-generic/add-generic.component.html [hunk 2]
```diff
@@ -46,13 +46,13 @@ <h1 class="center">
           class="box-content-row"
           appStopClick
           appBlurClick
-          (click)="selectType(cipherType.Identity)"
+          (click)="selectType(cipherType.Contact)"
```
Comment (Ldoppea) [inline apps/browser/src/cozy/components/add-generic/add-generic.component.html:49]: Based on who will merge first, those lines will be in conflict with https://github.com/cozy/cozy-keys-browser/pull/169/files#diff-d432dbbe8e3239f3af570459dc46515f8a53a17feaf2c31a8f0d95813e515833L44-L56

File: apps/browser/src/vault/popup/components/vault/view.component.ts [hunk 3]
```diff
@@ -190,6 +190,9 @@ export class ViewComponent extends BaseViewComponent {
     if (this.cipher.type === CipherType.Paper) {
       this.openWebApp();
       return;
+    } else if (this.cipher.type === CipherType.Contact) {
+      const hash = `${this.cipher.id}/edit`;
+      window.open(this.cozyClientService.getAppURL("contacts", hash));
```
Comment (Ldoppea) [inline apps/browser/src/vault/popup/components/vault/view.component.ts:195]: nit: took me a bit of time to understand the difference with the code inside of `openWebApp` and I was wondering why not calling `openWebApp`. Maybe extracting a method called `editInWebApp` and put next to `openWebApp` would ease understanding it faster. But this is pure nit.

**Review by Ldoppea — APPROVED** [review]

File: apps/browser/src/vault/popup/components/vault/view.component.ts (hunk 3 above)
Comment (zatteo) [reply apps/browser/src/vault/popup/components/vault/view.component.ts:195]: Agree

File: apps/browser/src/cozy/components/add-generic/add-generic.component.html (hunk 2 above)
Comment (zatteo) [reply apps/browser/src/cozy/components/add-generic/add-generic.component.html:49]: Sorry 😛

**Pull request merged**